# Machine Learning in Python
This notebook is primarily for learning the fundamentals and basics of Machine learning in Python

***

### Mean, Median, Mode

When looking at a dataset, what can we learn?
Primarily, in data science, there are 3 values that interest us:
- Mean - The Arithmetic average value
- Median - The mid-point value ($Q_2$)
- Mode - The most common value

In [18]:
import numpy as np
from scipy import stats

data = np.random.randint(0,100, size=50)
print(data)

[30 93 94  2 82 73 65 31 72 96 44 38  5 40 40  4 56 17 50 93 45 50 97 16
 98 75 91  4 56 23 30 85 51 45 36 38 10 36 89 20 98 53 78 76 37 17 73 27
 45 96]


Looking at the data above, we can use some tools from numpy and scipy to calculate some values.

In [19]:
# Finding Mean
np.mean(data)

52.4

In [20]:
# Finding Median
np.median(data)

47.5

In [21]:
# Finding Mode
stats.mode(data)

ModeResult(mode=45, count=3)

***

### Standard Deviation
Standard Deviation describes how spread out the values are.
- A low standard deviation means that most of the values are close to the mean.
- A high standard deviation means that the values are spread out over a wider range from the mean.

Using the same randomly generated set, we can calculate the standard deviation

In [26]:
np.std(data).round(2)

29.58

Which means most of the data values are within the range of 29.58 from the mean value.